calypso/service.go

```diff
@@ -711,36 +757,50 @@ func (s *Service) NewProtocol(tn *onet.TreeNodeInstance, conf *onet.GenericConfi
 
 		// Wait for DKG in reshare mode to end
 		go func(id byzcoin.InstanceID) {
+			// TODO: properly propagate errors during execution of DKG protocol
+			// (see dedis/cothority#2320)
 			<-setupDKG.Finished
+
+			setValidPeers := func(r *onet.Roster) {
+				s.SetValidPeers(s.NewPeerSetID(id[:]), r.List)
+			}
+
+			s.storage.Lock()
+
 			shared, dks, err := setupDKG.SharedSecret()
 			if err != nil {
+				setValidPeers(s.storage.Rosters[id])
+				s.storage.Unlock()
 				log.Error(err)
 				return
 			}
 
-			s.storage.Lock()
 			// If we had an old share, check the new share before saving it.
 			if s.storage.Shared[id] != nil {
 				// Check the secret shares are different
 				if shared.V.Equal(s.storage.Shared[id].V) {
+					setValidPeers(s.storage.Rosters[id])
 					s.storage.Unlock()
 					log.Error("the reshared secret is the same")
 					return
 				}
 
 				// Check the public key remains the same
 				if !shared.X.Equal(s.storage.Shared[id].X) {
+					setValidPeers(s.storage.Rosters[id])
 					s.storage.Unlock()
 					log.Error("the reshared public point is different")
 					return
 				}
 			}
 			s.storage.Shared[id] = shared
 			s.storage.DKS[id] = dks
+			s.storage.Rosters[id] = roster
+			setValidPeers(roster)
```

Comment:
My wrong understanding of the calypso re-sharing: the new roster should already be set here...

Reply:
Removed `setValidPeers()` call